node/pkg/governor/governor.go

```diff
@@ -924,11 +965,72 @@ func computeValue(amount *big.Int, token *tokenEntry) (uint64, error) {
 	return value, nil
 }
 
+// tryAddFlowCancelTransfer adds an inverse transfer to the destination chain entry. This occurs only if the transfer's asset
+// is allow-listed, and if the source and destination chain are connected via an enabled "pipe" (allow-list of chain ID pairs).
+// A return value of false means that the transfer will not do flow cancelling, i.e. it will not reduce the destination chain's
+// Governor usage.
+func (gov *ChainGovernor) tryAddFlowCancelTransfer(transfer *transfer) (bool, error) {
+	originChain := transfer.dbTransfer.OriginChain
+	originAddr := transfer.dbTransfer.OriginAddress
+	hash := transfer.dbTransfer.Hash
+	target := transfer.dbTransfer.TargetChain
+	emitter := transfer.dbTransfer.EmitterChain
+
+	pipe := &pipe{emitter, target}
+
+	// Check whether the source and destination chain are in the allow-list for flow cancelling.
+	if !gov.pipeCanFlowCancel(pipe) {
```

Comment:
Since this involves a for loop, would it be cheaper to do this after line 1001?

Reply:
The loop iterates over `gov.flowCancelPipes` which only contains one element now, and I think it's unlikely to grow much beyond that. So I don't think there will be much of a change in performance either way, unless Go has a big performance overhead for for loops that I don't know about.